test/concourse/steps/release_test.py

```diff
@@ -0,0 +1,257 @@
+                release_version=release_version,
+                version_operation=version_operation,
+                prerelease_suffix=prerelease_suffix,
+            )
+        return constructor
+
+    def test_validation(self, examinee):
+        examinee().validate()
+
+    def test_validation_fail_on_missing_version_file(self, examinee, tmpdir_str):
+        # create a path in the pytest test folder
+        test_path = os.path.join(tmpdir_str, 'made', 'up', 'path')
+        with pytest.raises(ValueError):
+            examinee(repository_version_file_path=test_path).validate()
+
+    def test_validation_fail_on_nonexistent_repo_dir(self, examinee, tmpdir_str):
+        # create a path in the pytest test folder
+        test_path = os.path.join(tmpdir_str, 'made', 'up', 'path')
+        with pytest.raises(ValueError):
+            examinee(repo_dir=test_path).validate()
+
+
+class TestPublishReleaseNotesStep(object):
+    @pytest.fixture()
+    def examinee(self, tmpdir_str):
+        def constructor(
+            github_helper=MagicMock(),
+            githubrepobranch=MagicMock(),
+            repo_dir=tmpdir_str,
+            release_version='1.0.0',
+        ):
+            return concourse.steps.release.PublishReleaseNotesStep(
+                github_helper=github_helper,
+                githubrepobranch=githubrepobranch,
+                repo_dir=repo_dir,
+                release_version=release_version,
+            )
+        return constructor
+
+    def test_validation(self, examinee):
+        examinee().validate()
+
+    def test_validation_fail_on_nonexistent_repo_dir(self, examinee, tmpdir_str):
+        # create a path in the pytest test folder
+        test_path = os.path.join(tmpdir_str, 'made', 'up', 'path')
+        with pytest.raises(ValueError):
+            examinee(repo_dir=test_path).validate()
+
+    def test_validation_fails_on_invalid_semver(self, examinee):
+        with pytest.raises(ValueError):
+            examinee(release_version='made_up_version').validate()
+
+
+class TestCleanupDraftReleaseStep(object):
+    @pytest.fixture()
+    def examinee(self, tmpdir_str, existing_file_path):
+        def constructor(
+            github_helper=MagicMock(),
+            release_version='1.0.0',
+        ):
+            return concourse.steps.release.CleanupDraftReleaseStep(
+                github_helper=github_helper,
+                release_version=release_version,
+            )
+        return constructor
+
+    def test_validation(self, examinee):
+        examinee().validate()
+
+    def test_validation_fails_on_invalid_semver(self, examinee):
+        with pytest.raises(ValueError):
+            examinee(release_version='made_up_version').validate()
+
+
+class TestSlackReleaseStep(object):
+    @pytest.fixture()
+    def examinee(self):
+        def constructor(
+            slack_cfg_name='test_config',
+            slack_channel='test_channel',
+            githubrepobranch=MagicMock(),
```

Comment:
please create real objects here. there is no need to use a mock, since this is just a plain python object contain some strings, is it not?